app/controllers/api/v1/markers/pins_controller.rb

```diff
@@ -0,0 +1,10 @@
+module Api::V1::Markers
+  class PinsController < ApplicationController
+
+    def index
+      pin = Pin.find_by(marker_id: params[:marker_id])
+      render json: pin
```

Comment:
index action supposed to response with an array but here is an object, you should use url /api/pins/:id which routes to action show